contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {CoreRef} from "../refs/CoreRef.sol";
+import {TribeRoles} from "./../core/TribeRoles.sol";
+import {RateLimited} from "./RateLimited.sol";
+import {IMultiRateLimited} from "./IMultiRateLimited.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+/// @title contract for putting a rate limit on how fast an address can perform an action e.g. Minting
+/// there are two buffers, one buffer which is each individual addresses's current buffer,
+/// and then there is a global buffer which is the buffer that each individual address must respect as well
+/// @author Fei Protocol
+contract MultiRateLimited is RateLimited, IMultiRateLimited {
+    using SafeCast for *;
+
+    /// @notice the struct containing all information per rate limited address
+    struct RateLimitData {
+        uint32 lastBufferUsedTime;
+        uint144 bufferCap;
+        uint112 rateLimitPerSecond;
+        uint144 bufferStored;
+    }
+
+    /// @notice rate limited address information
+    mapping (address => RateLimitData) public rateLimitPerAddress;
+
+    /// @notice max rate limit per second allowable by non governor
+    uint256 public maxRateLimitPerSecond;
+
+    /// @notice max buffer cap allowable by non governor
+    uint256 public maxBufferCap;
```

Comment:
Sorry for the back and forth here, but I suggest changing this to be more descriptive. Even after spending a couple of minutes looking over the code, I can't tell if this maximum applies to the global buffer or the per-contract buffers.

Reply:
the comment literally says what the variable is.

Reply:
No it doesn't? Again: "I can't tell if this maximum applies to the global buffer or the per-contract buffers"